modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionServiceImpl.java

```diff
@@ -442,9 +443,10 @@ private AsyncDataCursor<InternalSqlRow> executeExecutablePlan(
         QueryTransactionWrapper txWrapper = txContext.explicitTx();
 
         if (txWrapper == null) {
-            // underlying table will initiate transaction by itself, but we need stub to reuse
-            // TxAwareAsyncCursor
-            txWrapper = NoopTransactionWrapper.INSTANCE;
+            txWrapper = plan.transactional()
```

Comment:
the only "non-transactional" plan is `SelectCountPlan` which is more like temporal solution. I would prefer not to introduce `transactional` property on `ExecutablePlan` interface, and get rid of `NoopTransactionWrapper` instead

Reply:
Done